op-service/ioutil/gzip.go

```diff
@@ -9,26 +9,57 @@ import (
 	"strings"
 )
 
+// gzipReadCloser is a struct that closes both the gzip.Reader and the underlying io.Closer.
+type gzipReadCloser struct {
+	io.ReadCloser
+	closer io.Closer
+}
+
+// Close closes both the gzip.Reader and the underlying reader.
+func (g *gzipReadCloser) Close() error {
+	if err := g.ReadCloser.Close(); err != nil {
+		return err
+	}
+	return g.closer.Close()
```

Comment:
I suspect we should try to close both even if one fails, so should use something like:
```suggestion
    return errors.Join(g.ReadCloser.Close(), g.closer.Close())
```

`errors.Join` will ignore params that are `nil` so if both return `nil` this still returns `nil` overall.